### ISSUE TITLE
Invalid MIT-MAGIC-COOKIE-1 key

### ISSUE THREAD

**Issue by rasmus91** [flatpak/flatpak#3442]
## OS and Version
elementary OS 5.1 (Hera)
## Flatpak version
 Flatpak 1.6.2
## Description of the problem
Whenever i try to run a flatpak, e.g. `flatpak run <app-name>` I simply get a message that i have an invalid magic cookie, here's the example with slack

```flatpak run com.slack.Slack 
Gtk-Message: 21:30:41.689: Failed to load module "pantheon-filechooser-module"
Invalid MIT-MAGIC-COOKIE-1 key
(slack:2): Gtk-WARNING **: 21:30:41.690: cannot open display: :99.0
[fake-sandbox: zypak-sandbox] No data could be read (host died?)
[fake-sandbox: zypak-sandbox] Quitting Zygote... 
```

## Steps to reproduce
I've no idea why or how it does this. This time, It happened when I had booted my laptop at home. 
I clicked the icon, but i just got the spinning cursor. So then i tried running from terminal and I got the above message.

**Comment by rasmus91** [comment]
Right! So I tried a few things in the terminal, then went to do reboot bot it told me that user `nobody` was logged into the system...

So i went to good ol' FF and searched for that. Turns out `nobody` is sometimes used by systemd (or its units?) to run certain services, then i recalled i installed `ofono-phonsim-autostart` today, and while starring at the terminal i noticed it creating a systemd file. So I tried apt removing that package, rebooting and behold; flatpaks work on my system again.

I don't know if its the specific service that mucks it up, or the nobody user in general, can someone else try with the `ofono-phonesim-autostart`  package, and then maybe tell if there's another package that uses user `nobody`, that i can test with, so we can ascertain where exactly the problem lies? Because this seems to somehow have messed with the X session

**Comment by TingPing** [comment]
Maybe the service is overwriting the `XAUTHORITY` env var

**Comment by rexkerr** [comment]
Had the same problem with being unable to launch flatpak apps that used X, with a similar MIT-MAGIC-COOKIE-1 error.  Searching brought me here, and sure enough I'd installed the ofono-phonesim-autorestart package today, and removing it fixed the problem.
